### DISCUSSION TITLE
Wear OS: Ignore DND mode for glucose alerts

### DISCUSSION THREAD

**Discussion by xannasavin** [NightscoutFoundation/xDrip#2224] [Ideas]
Sorry for opening an issue prior to this.

**Your environment**

- What version of xDrip: 2022.05.26
- What type of hardware data source: Dexcom G6
- Ticwatch Pro 3 GPS, Huawei P30 Pro

**Expected behavior**

- I do activate the DND mode at night for my watch.
- The alarms for high or low glucose should vibrate.

**Actual behavior**

In the past I solved this issue with the "Feel the wear" app, but since it is outdated it's no more possible to install it on the watch.
The watch is, as said, at night in DND mode, so I don't get disturbed while sleeping.
However, the alarms for high or low glucose are obviously essential.
There should be a way to allow those notifications to bypass the DND mode on the watch.
"Feel the wear" made it also "somehow" possible to ignore DND mode, so there should be a solution.
I can not let my phone do the alarms, since that would wake up others, which I also do not want.

**Comment by Navid200** [comment]
With respect to the phone waking everyone else up, I wonder if you have considered the ascending volume profile.  It is an option:  https://navid200.github.io/xDrip/docs/Ascending-volume-profile.html
In short, the alert starts at low volume.  If the phone is next to you, and if the others are not extremely light sleepers, you should be able to snooze the alert before its volume raises to maximum waking up everyone else.

@psonnera  Any thoughts on Wear OS?

**Comment by xannasavin** [comment]
thanks for that hint @Navid200 , but in fact I am a extreme light sleeper that's why I wear earplugs at night - you see where this is going? :D Everyone would be awake before I would be... 
There would be a lot of situations also during the day, where the watch would be on DND like meetings, classes or theater. 
So I would not limit the use case to sleeping.

**Reply by psonnera** [reply]
@xannasavin do you have vibration on your watch? For my son I enabled that with a silent ringtone on the phone.

**Reply by xannasavin** [reply]
yes I have vibration on my watch - how did you configure this? and does it work though DND is active on the watch?

**Reply by psonnera** [reply]
Use a silent ringtone on xDrip+ alarm(s), enable vibration and bypass DND. This will make the phone and watch vibrate but not ring. I didn't enable DND on the watch. Used a Huawei Watch 2 LTE, just disabled all phonecalls sounds. Using it for a kid I had very few apps installed on it (actually only xDrip+).

**Reply by xannasavin** [reply]
I gave this a quick try - thank you!
 When the setting "overwrite silent mode" is enabled in the alarm settings, the watch vibrates when in DND mode. However, normally also the theater mode is active at night, which this trick can not overcome. Besides of the need of a silent ring tone this trick is nice, but unfortunately does not solve my issue completely.

**Comment by xannasavin** [comment]
I managed to get hands on the wear apk for feel the wear and sideloaded it to my watch - so this works again.
maybe it's still something that might be valuable to add to xDrip as well, that it can bypass dnd and theater mode for alerts.
